src/dungeon_ai_leader.c

```diff
@@ -33,3 +109,268 @@ Entity* GetLeaderIfVisible(Entity *pokemon)
+        info->attacking = TRUE;
+    }
+    bVar14 = FALSE;
+    info->unk14A = 0;
+    gUnknown_202F221 = 0;
+    gUnknown_202F222 = 0;
+    switch(info->action.action) {
+        case ACTION_WALK:
+            if(info->immobilizeStatus == STATUS_SHADOW_HOLD)
+            {
+                goto _282;  
+            }
+            else if(info->immobilizeStatus == STATUS_CONSTRICTION)
+            {
+                goto _282;  
+            }
+            else if(info->immobilizeStatus == STATUS_INGRAIN)
+            {
+                goto _282;  
+            }
+            else if(info->immobilizeStatus == STATUS_WRAP)
+            {
+                goto _282;  
+            }
+            else if(info->immobilizeStatus == STATUS_WRAPPED)
+            {
+            _282:
+                info->action.action = ACTION_NOTHING;
+                info->waiting = TRUE;
+            }
+            else {
+                bVar5 = FALSE;
+                pos1.x = (entity->pos).x + gAdjacentTileOffsets[(info->action).direction].x;
+                pos1.y = (entity->pos).y + gAdjacentTileOffsets[(info->action).direction].y;
+                if ((!(info->flags & MOVEMENT_FLAG_SWAPPING_PLACES_PETRIFIED_ALLY)) &&
+                    (!CanMoveInDirection(entity,(info->action).direction))) {
+                    bVar5 = TRUE;
+                }
+                if (bVar5) {
+                    info->action.action = ACTION_NOTHING;
+                    info->waiting = TRUE;
+                    break;
+                }
+                else
+                {
+                    PStack_24.x = entity->pos.x;
+                    PStack_24.y = entity->pos.y;
+                    sub_80694C0(entity,pos1.x,pos1.y,0);
+                    sub_8074FB0(entity,(info->action).direction,&PStack_24);
+                    if (((IQSkillIsEnabled(entity, IQ_SUPER_MOBILE)) && (info->transformStatus != STATUS_MOBILE)) &&
+                        (!HasHeldItem(entity,ITEM_MOBILE_SCARF))) {
+                        sub_804AE08(&entity->pos);
+                    }
+                    bVar14 = TRUE;
+                    break;
+                }
+            }
+            break;
+        case ACTION_USE_MOVE_AI:
+            sub_807CABC(entity);
+            break;
+        case ACTION_STAIRS:
+            if ((gDungeon->dungeonLocation.id == DUNGEON_METEOR_CAVE) && (!gDungeon->deoxysDefeat)) {
+                SendMessage(entity,*gUnknown_80FA5B4); // It's impossible to go down the stairs now!
+            }
+            else
+            {
+                sub_804267C();
+                gDungeon->unk2 = 1;
+                gUnknown_202F32C = (info->action).direction;
+            }
+            break;
+        case ACTION_EAT_BERRY_SEED_DRINK:
+        case 0xe:
+        case ACTION_USE_TM:
+            sub_8067110(entity);
+            break;
+        case ACTION_USE_ORB:
+            sub_807CB3C(entity);
```

Comment:
We have a few of these decomped already but not sure what to name them..

`HandleACTIONNAMEAction`? ->  `HandleUseOrbAction` & `HandleGiveItemAction`

Reply:
I'm good with that naming convention.